lib/controllers/pr-reviews-controller.js

```diff
@@ -0,0 +1,155 @@
+    }
+  }
+
+  handleError = error => {
+    /* istanbul ignore if */
+    if (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+    } else {
+      this._attemptToLoadMoreReviews();
+    }
+  }
+
+  _loadMoreReviews = () => {
+    this.props.relay.loadMore(PAGE_SIZE, this.handleError);
+  }
+
+  render() {
+    if (!this.props.pullRequest || !this.props.pullRequest.reviews) {
+      return null;
+    }
+
+    const commentThreads = Object.keys(this.state).reverse().map(rootCommentId => {
+      return {
+        rootCommentId,
+        comments: this.state[rootCommentId],
+      };
+    });
+
+    /** Dealing with comment threading...
+      *
+      * Threads can have comments belonging to multiple reviews.
+      * We need a nested pagination container to fetch comment pages.
+      * Upon fetching new comments, the `collectComments` method is called with all comments fetched for that review.
+      * Ultimately we want to group comments based on the root comment they are replies to.
+      * `renderCommentFetchingContainers` only fetches data and doesn't render any user visible DOM elements.
+      * `PullRequestReviewCommentsView` renders the aggregated comment thread data.
+    * */
+    return (
+      <Fragment>
+        {this.renderCommentFetchingContainers()}
+        <PullRequestReviewCommentsView commentThreads={commentThreads} {...this.props} />
+      </Fragment>
+    );
+  }
+
+  renderCommentFetchingContainers() {
+    return this.props.pullRequest.reviews.edges.map(({node: review}) => {
+      return (
+        <PullRequestReviewCommentsContainer
+          key={review.id}
+          review={review}
+          collectComments={this.collectComments}
+        />
+      );
+    });
+  }
+
+  collectComments = ({reviewId, submittedAt, comments, fetchingMoreComments}) => {
+    this.reviewsById.set(reviewId, {submittedAt, comments, fetchingMoreComments});
+    const stillFetchingReviews = this.props.relay.hasMore();
+    if (!stillFetchingReviews) {
+      const stillFetchingComments = [...this.reviewsById.values()].some(review => review.fetchingMoreComments);
+      if (!stillFetchingComments) {
+        this.groupCommentsByThread();
+      }
+    }
+  }
+
+  groupCommentsByThread() {
+    // we have no guarantees that reviews will return in order so sort them by date.
+    const sortedReviews = [...this.reviewsById.values()].sort(this.compareReviewsByDate);
+
+    // react batches calls to setState and does not update state synchronously
+    // therefore we need an intermediate state so we can do checks against keys
+    // we have just added.
+    const state = {};
+    sortedReviews.forEach(({comments}) => {
+      comments.edges.forEach(({node: comment}) => {
+        if (!comment.replyTo) {
+          state[comment.id] = [comment];
+        } else {
+          // TODO: look at this more closely...
+          // When comment being replied to is outdated...?? Not 100% sure...
```

Comment:
I tried replying to an outdated comment, and it did not show up in the pull request view.  So yeah, I have no idea how we'd get into this state.

@kuychaco are there other ways you'd suggest we investigate this?  I'm not sure what else to do.

Reply:
Maybe when replying to a deleted or hidden comment?

Reply:
okay, so I tried:

- replying to a deleted comment
- replying to a resolved comment
- replying to a hidden comment

None of these triggered that condition for line 132.  However, I did notice we are not respecting hidden comments.  That seems like a ship blocker to me since we should not be showing users comments that have been marked as abusive or spammy.